docs/Game/Unreal/getting-started/02-sending-transactions.md

```diff
@@ -0,0 +1,346 @@
+
+### Response
+
+Successful or Unsuccessful.
+
+Get Data
+-------
+-------
+
+**GetData** is used to retrieve data associated with a particular wallet and user from a contract.
+
+### URL
+
+http://45.77.189.28:5000/wallet/call/method
+
+### Parameters
+
+`device_id`, 
+`contract_address`, 
+`abi_hash`, 
+`method`, 
+`args`
+
+### Response
+
+The data is returned. 
+
+```cpp
+void UMirageClient::GetData(FString contract, FString abi_hash, FString method, FString args, FMirageDelegate Result)
+{
+	http = &FHttpModule::Get();
+
+	TSharedRef<IHttpRequest, ESPMode::ThreadSafe> Request = http->CreateRequest();
+	Request->OnProcessRequestComplete().BindLambda([Result, this](FHttpRequestPtr Request, FHttpResponsePtr Response, bool bWasSuccessful)
+		{
+			TSharedPtr<FJsonObject> JsonObject;
+			TSharedRef<TJsonReader<>> Reader = TJsonReaderFactory<>::Create(Response->GetContentAsString());
+			Result.ExecuteIfBound(Response->GetContentAsString());
+		});
+
+	FString url = baseUrl + "call/method";
+
+	Request->SetURL(url);
+	Request->SetVerb("POST");
+	Request->SetHeader(TEXT("User-Agent"), "X-MirageSDK-Agent");
+	Request->SetHeader("Content-Type", TEXT("application/json"));
+	Request->SetContentAsString("{\"device_id\": \"" + deviceId + "\", \"contract_address\": \"" + contract + "\", \"abi_hash\": \"" + abi_hash + "\", \"method\": \"" + method + "\", \"args\": \"" + args + "\"}");
+	Request->ProcessRequest();
+}
+```
+
+Send ABI
+-----
+-----
+
+**SendABI** is used to send a request and retrieve the ABI hash generated by the backend
+
+### URL
+
+http://45.77.189.28:5000/abi
+
+### Parameters
+
+`abi`
+
+### Response
+
+The ABI hash is returned. 
+
+```cpp
+
+void UMirageClient::SendABI(FString abi, FMirageDelegate Result)
+{
+	http = &FHttpModule::Get();
+
+	TSharedRef<IHttpRequest, ESPMode::ThreadSafe> Request = http->CreateRequest();
+	Request->OnProcessRequestComplete().BindLambda([Result, this](FHttpRequestPtr Request, FHttpResponsePtr Response, bool bWasSuccessful)
+		{
+			TSharedPtr<FJsonObject> JsonObject;
+			TSharedRef<TJsonReader<>> Reader = TJsonReaderFactory<>::Create(Response->GetContentAsString());
+			FString data = Response->GetContentAsString();
+			
+			if (FJsonSerializer::Deserialize(Reader, JsonObject))
+			{
+				Result.ExecuteIfBound(JsonObject->GetStringField("abi"));
+			}
+		});
+
+	FString url = baseUrl + "abi";
+
+	Request->SetURL(url);
+	Request->SetVerb("POST");
+	Request->SetHeader(TEXT("User-Agent"), "X-MirageSDK-Agent");
+	Request->SetHeader("Content-Type", TEXT("application/json"));
+
+	const TCHAR* find = TEXT("\"");
+	const TCHAR* replace = TEXT("\\\"");
+	FString body = FString("{\"abi\": \"" + abi.Replace(find, replace, ESearchCase::IgnoreCase) + "\"}");
+	
+	Request->SetContentAsString(*body);
+	Request->ProcessRequest();
+}
+```
+
+Sign Message
+---
+---
+
+**SignMessage** is used to send a request that requires confirmation in MetaMask. The session saved during **GetClient** will be used to open MetaMask. 
```

Comment:
`SignMessage` and `GetClient`? 

saved during calling `GetClient`?

Reply:
done